### PR TITLE
Redis TokenBucket should use the interface not the client struct

### PR DESCRIPTION
**Pull request by VivaLaPanda**

### CODE REVIEW

File: tokenbucket.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 	"sync"
 	"time"
 
-	"github.com/go-redis/redis"
+	"github.com/go-redis/redis/v8"
```
Comment (mennanov) [inline tokenbucket.go:10]: Is there a reason the version `v8` is used instead of `v9` which is already available?

**Review by mennanov — CHANGES_REQUESTED** [review]
It looks like not all references are updated to use `redis.Cmdable` (see the failing tests).
Please, fix that.

File: tokenbucket.go (hunk 1 above)
Comment (VivaLaPanda) [reply tokenbucket.go:10]: The code that I was looking at using this on was using v8 so I went for that, if you want to bump all the way to 9 that seems fine

**Comment by mennanov** [comment]
Please, update your branch: it's running behind the main branch

**Review by mennanov — CHANGES_REQUESTED** [review]
Please update this branch and resolve all the conflicts

**Comment by mennanov** [comment]
@VivaLaPanda do you think this PR is still relevant?  A somewhat similar PR https://github.com/mennanov/limiters/pull/15 has been merged recently.

**Comment by mennanov** [comment]
Closing this PR as inactive.

**Pull request closed without merging**